docs/orleans/quickstarts/build-your-first-orleans-app.md

```diff
@@ -0,0 +1,325 @@
+
+- `IGrainWithGuidKey`
+- `IGrainWithIntegerKey`
+- `IGrainWithStringKey`
+- `IGrainWithGuidCompoundKey`
+- `IGrainWithIntegerCompoundKey`
+
+For this quickstart, you'll use the `IGrainWithStringKey`, since strings are a logical choice for working with URL values and short codes.
+
+Orleans grains can also use a custom interface to define their methods and properties. The URL shortener grain interface should define two methods:
+
+- A `SetUrl` method to persist the original and shortened URLs.
+- A `GetUrl` method to retrieve the original URL using the shortened URL.
+
+1. Append the following interface definition to the bottom of the _Program.cs_ file.
+
+    ```csharp
+    public interface IUrlShortenerGrain : IGrainWithStringKey
+    {
+        Task SetUrl(string shortenedRouteSegment, string fullUrl);
+        Task<string> GetUrl();
+    }
+    ```
+
+1. Create a `UrlShortenerGrain` class using the following code. This class inherits from the `Grain` class provided by Orleans and implements the `IUrlShortenerGrain` interface you created. The class also uses the `IPersistentState` interface of Orleans to manage reading and writing state values for the URLs to the configured silo storage.
+
+    ```csharp
+    public class UrlShortenerGrain : Grain, IUrlShortenerGrain
+    {
+        private readonly IPersistentState<KeyValuePair<string, string>> _cache;
+    
+        public UrlShortenerGrain(
+            [PersistentState(
+                stateName: "url",
+                storageName: "urls")]
+                IPersistentState<KeyValuePair<string, string>> state)
+        {
+            _cache = state;
+        }
+    
+        public async Task SetUrl(string shortenedRouteSegment, string fullUrl)
+        {
+            _cache.State = new KeyValuePair<string, string>(shortenedRouteSegment, fullUrl);
+            await _cache.WriteStateAsync();
+        }
+    
+        public Task<string> GetUrl()
+        {
+            return Task.FromResult(_cache.State.Value);
+        }
+    }
+    ```
+
+## Create the endpoints
+
+Next, create two endpoints to utilize the Orleans grain and silo configurations:
+
+- A `/shorten/{*url}` endpoint to handle creating and storing a shortened version of the URL. The original, full URL is provided as a path parameter, and the shortened URL is returned to the user for later use.
+- A `/go/{*shortenedUrl}` endpoint to handle redirecting users to the original URL using the shortened URL that is supplied as a parameter.
+
+Inject the <xref:Orleans.IGrainFactory> interface into both endpoints. Grain Factories enable you to retrieve and manage references to individual grains that are stored in silos. Append the following code to the end of the _Program.cs_ file before the `app.Run()` method call:
+
+```csharp
+app.MapGet("/shorten/{*path}",
```

Comment:
Should this be
```suggestion
app.MapPost("/shorten/{*path}",
```
rather than HTTP-GET?
And then `Results.CreateAt` instead of `OK`?

Reply:
Apologies for missing this. I originally left this as a GET to make it easier to test right in the browser, otherwise the user has to author a post request using a tool or extension.  However, I can also change this to a POST to ensure it follows best practices.